wordpress-mobile/WordPress-iOS
# Try/new rich text

Refs #5594

This PR introduces a new rich text solution in the Reader’s post detail as a replacement to DTCoreText. 

WPRichTextView is replaced in the reader detail by the new WPRichContentView, a subclass of UITextView that relies on Apple’s TextKit framework for rendering HTML in a UITextView. Special treatment is given to blockquote tags, images, and other “embedded” HTML content. 

The new WPRichText classes are based on previous effort in WPTextKit — greatly improved from their WPTextKit incarnation and incorporated directly into WPiOS instead of as a separate library. 

We'll address making the same change in reader comments, removing the DTCoreText pod, and renaming WPRichContentView to WPRichTextView in future PRs.

To test: 
Take the reader detail view for a spin and check for the following things:
- The layout is correct.
- Images are rendered correctly. 
- Images wrapped in a link open the link when tapped
- Images that are not wrapped in a link show the image when tapped.
- Images smaller than the width of the text content are centered.
- Images shorter than the line height are rendered inline.
- Videos (youtube, vimeo, etc.) embedded as iframe are rendered correctly, and play when tapped.
- Confirm that block quotes are indented, italicized, etc.

Confirm that tapping a link opens the in-app browser.

Confirm that posts with lots and lots of images do not suffer from undue lag when loading in the detail.

Confirm that Discover posts have the correct layout for their attribution.


Needs review: @jleandroperez, @koke, could I trouble you both for this one?

cc @frosty @kurzee @bummytime as this is likely relevant to your interests as well.
